pkg/container/docker/image.go

```diff
@@ -17,10 +17,52 @@ limitations under the License.
 package docker
 
 import (
+	"fmt"
+	"strings"
+
 	"github.com/pkg/errors"
 	"sigs.k8s.io/kind/pkg/exec"
 )
 
+// SplitImage splits an image into (registry,tag) following these cases:
+//
+//  alpine -> (alpine, latest)
+//
+//  alpine:latest -> (alpine, latest)
+//
+//  alpine@sha256:28ef97b8686a0b5399129e9b763d5b7e5ff03576aa5580d6f4182a49c5fe1913 -> (alpine, latest@sha256:28ef97b8686a0b5399129e9b763d5b7e5ff03576aa5580d6f4182a49c5fe1913)
+//
+//  alpine:latest@sha256:28ef97b8686a0b5399129e9b763d5b7e5ff03576aa5580d6f4182a49c5fe1913 -> (alpine, latest@sha256:28ef97b8686a0b5399129e9b763d5b7e5ff03576aa5580d6f4182a49c5fe1913)
+//
+// NOTE: for our purposes we consider the sha to be part of the tag, and we
+// resolve the implicit :latest
+func SplitImage(image string) (registry, tag string, err error) {
+	// we are looking for ':' and '@'
+	firstColon := strings.IndexByte(image, 58)
+	firstAt := strings.IndexByte(image, 64)
+
+	// there should be a registry before the tag, and @/: should not be the last
+	// character, these cases are assumed not to exist by the rest of the code
+	if firstColon == 0 || firstAt == 0 || firstColon+1 == len(image) || firstAt+1 == len(image) {
+		return "", "", fmt.Errorf("unexpected image: %q", image)
```

Comment:
will do a sweep and convert to pkg/errors later. would rather focus on getting the fix in and I don't think this error will be difficult to locate.